Providers/Profile.swift

```diff
@@ -1208,7 +1229,8 @@ public class BrowserProfile: Profile {
         }
 
         // This SHOULD NOT be called directly: use syncSeveral instead.
-        private func syncWith(synchronizers: [(EngineIdentifier, SyncFunction)]) -> Deferred<Maybe<[(EngineIdentifier, SyncStatus)]>> {
+        private func syncWith(synchronizers: [(EngineIdentifier, SyncFunction)],
+                              statsSession: SyncOperationStatsSession) -> Deferred<Maybe<[(EngineIdentifier, SyncStatus)]>> {
             guard let account = self.profile.account else {
```

Comment:
I suspect that we should be capturing `self.profile.account` in `syncSeveral`, doing this check there, and passing it as an argument to `syncWith`, alongside the stats session. That allows us to construct the `statsSession` from the account directly in `syncSeveral`, which will make it less likely that someone will forget to set its fields…